lib/daru/io/importers/csv.rb

```diff
@@ -77,33 +84,53 @@ def initialize(path, headers: nil, col_sep: ',', converters: :numeric,
         def call
           # Preprocess headers for detecting and correcting repetition in
           # case the :headers option is not specified.
+
+          @file_string = process_compression
+
           hsh =
             if @headers
               hash_with_headers
             else
               hash_without_headers.tap { |hash| @daru_options[:order] = hash.keys }
             end
+
           Daru::DataFrame.new(hsh,@daru_options)
         end
 
         private
 
         def hash_with_headers
           ::CSV
-            .parse(open(@path), @options)
+            .parse(@file_string, @options)
             .tap { |c| yield c if block_given? }
-            .by_col.map { |col_name, values| [col_name, values] }.to_h
+            .by_col
+            .map do |col_name, values|
+              next [col_name, []] if values.nil?
+              [col_name, values[@skiprows..-1]]
+            end
+            .to_h
         end
 
         def hash_without_headers
           csv_as_arrays =
             ::CSV
-            .parse(open(@path), @options)
+            .parse(@file_string, @options)
             .tap { |c| yield c if block_given? }
             .to_a
           headers       = ArrayHelper.recode_repeated(csv_as_arrays.shift)
-          csv_as_arrays = csv_as_arrays.transpose
-          headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
+          csv_as_arrays = csv_as_arrays[@skiprows..-1].transpose
+          headers
+            .each_with_index
+            .map do |h, i|
+              next [h, []] if csv_as_arrays[i].nil?
+              [h, csv_as_arrays[i]]
+            end
+            .to_h
+        end
+
+        def process_compression
+          return ::Zlib::GzipReader.new(open(@path)).read if @compression == :gzip
+          open(@path)
         end
```

Comment:
Is it fine to compute `.gzip` readings in the `from_csv` method itself, or would it be neater to have `Importers::CSVGZip` inheriting `Importers::CSV`? I have the same doubt regarding exporter (PR #34) as well.